service/history/replication/stream_receiver.go

```diff
@@ -257,26 +319,98 @@ func (r *StreamReceiverImpl) processMessages(
 		if streamResp.Err != nil {
 			return streamResp.Err
 		}
-		tasks := r.taskConverter.Convert(
+		if err := r.validateAndSetReceiverMode(streamResp.Resp.GetMessages().Priority); err != nil {
+			r.logger.Error("ReplicationTask wrong receiver mode", tag.Error(err))
+			break // sender mode changed, exit loop and let stream reconnect to retry
+		}
+
+		if err = ValidateTasksHaveSamePriority(streamResp.Resp.GetMessages().Priority, streamResp.Resp.GetMessages().ReplicationTasks...); err != nil {
+			// This should not happen because source side is sending task 1 by 1. Validate here just in case.
+			r.logger.Error("ReplicationTask priority check failed", tag.Error(err))
+			// Todo: Change to write Tasks to DLQ. As resend task will not help here
+			break // exit loop and let stream reconnect to retry task
```

Comment:
Is this recoverable error? If not, could we add some metrics and alert?

Reply:
maybe just return error? would stream reconnect help?

Reply:
Changed to return stream error. So we will be notified if anything wrong.